server/src/test/java/org/elasticsearch/health/HealthServiceTests.java

```diff
@@ -159,30 +302,79 @@ public void testPreflightIndicatorFailureTriggersUnknownResults() {
         }
 
         {
-            List<HealthIndicatorResult> health = service.getHealth("slow_task_assignment", false);
+            List<HealthIndicatorResult> health = getHealthIndicatorResults(service, client, "slow_task_assignment");
             assertThat(health.size(), is(equalTo(1)));
             assertIndicatorIsUnknownStatus(health.get(0));
         }
 
         {
-            List<HealthIndicatorResult> health = service.getHealth("has_master", false);
+            List<HealthIndicatorResult> health = getHealthIndicatorResults(service, client, "has_master");
             assertThat(health.size(), is(equalTo(1)));
             assertThat(health.get(0), is(equalTo(hasMaster)));
         }
     }
 
+    private List<HealthIndicatorResult> getHealthIndicatorResults(HealthService service, NodeClient client, String indicatorName)
+        throws Exception {
+        AtomicReference<List<HealthIndicatorResult>> resultReference = new AtomicReference<>();
+        ActionListener<List<HealthIndicatorResult>> listener = new ActionListener<>() {
+            @Override
+            public void onResponse(List<HealthIndicatorResult> healthIndicatorResults) {
+                resultReference.set(healthIndicatorResults);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                throw new RuntimeException(e);
```

Comment:
Shall we use `fail` here too? In general in these case when an expected exception happened in a listener?

Reply:
Same reasoning as above.